### PR TITLE
StaticLLMPipeline: Decide when to enable NPUW_DQ_FULL property

### PR DESCRIPTION
**Pull request by smirnov-alexey**
Based on (yet to be) supported OV properties from the NPU Plugin enable NPUW_DQ_FULL.
releases/2024/5 mirror: https://github.com/openvinotoolkit/openvino.genai/pull/1272

Dependencies
* https://github.com/openvinotoolkit/openvino/pull/27678 needs to be merged first
* https://github.com/openvinotoolkit/openvino/pull/27789

### CODE REVIEW

File: src/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -71,7 +71,7 @@ target_include_directories(${TARGET_NAME}
 
 target_include_directories(${TARGET_NAME} SYSTEM PRIVATE "${safetensors.h_SOURCE_DIR}")
 
-target_link_libraries(${TARGET_NAME} PUBLIC openvino::runtime PRIVATE openvino::threading nlohmann_json::nlohmann_json jinja2cpp)
+target_link_libraries(${TARGET_NAME} PUBLIC openvino::runtime openvino::runtime::dev PRIVATE openvino::threading nlohmann_json::nlohmann_json jinja2cpp)
```
Comment (smirnov-alexey) [inline src/cpp/CMakeLists.txt:74]: Required for `ov::internal::supported_properties`. Is there any better way?

Comment (smirnov-alexey) [reply src/cpp/CMakeLists.txt]: Reworked

File: src/cpp/src/llm_pipeline_static.cpp [hunk 2]
```diff
@@ -466,7 +468,14 @@ std::optional<NPUDesc> extract_npu_descriptor(ov::Core& core) {
     }
     const auto arch = core.get_property("NPU", ov::device::architecture);
     const auto max_tiles = core.get_property("NPU", ov::intel_npu::max_tiles);
-    return std::make_optional(NPUDesc{arch, max_tiles});
+
+    bool compiler_dq = false;
+    const auto supported_internal_properties = core.get_property("NPU", ov::internal::supported_properties);
```
[the remark this answers is not in the document]
Comment (smirnov-alexey) [reply src/cpp/src/llm_pipeline_static.cpp]: Reworked with public properties

File: src/cpp/src/llm_pipeline_static.cpp [hunk 3]
```diff
@@ -466,7 +467,14 @@ std::optional<NPUDesc> extract_npu_descriptor(ov::Core& core) {
...
     const auto arch = core.get_property("NPU", ov::device::architecture);
     const auto max_tiles = core.get_property("NPU", ov::intel_npu::max_tiles);
-    return std::make_optional(NPUDesc{arch, max_tiles});
+
+    bool compiler_dq = false;
+    const auto device_caps = core.get_property("NPU", ov::device::capabilities);
+    if (std::find(device_caps.begin(), device_caps.end(),
+                  "COMPILER_DYNAMIC_QUANTIZATION") != device_caps.end()) {
+        compiler_dq = true;
+    }
```
Comment (dmatveev) [inline src/cpp/src/llm_pipeline_static.cpp:476]:
```suggestion
    const bool compiler_dq = caps.end() != std::find(caps.begin(), caps.end(), "COMPILER_DYNAMIC_QUANTIZATION")
```

File: src/cpp/src/llm_pipeline_static.cpp [hunk 4]
```diff
@@ -457,6 +457,7 @@ void merge_config_with(ov::AnyMap& lhs, const ov::AnyMap& rhs) {
 struct NPUDesc {
     std::string arch;
     int64_t max_tiles;
+    bool compiler_dq;
```
Comment (dmatveev) [inline src/cpp/src/llm_pipeline_static.cpp:460]: I think it is worth to have default values here, probably even incorrect (e.g., 0 as max_tiles)

**Comment by ilya-lavrenov** [comment]
Please, add `port to LTS` if required.
I've changed milestone to 2025.0 because target branch is master

**Comment by smirnov-alexey** [comment]
Property to be added in https://github.com/openvinotoolkit/openvino/pull/27789

**Pull request merged**